packages/spectacular-docs/docs/feature-testing/angular-testing-library.md

```diff
@@ -0,0 +1,575 @@
+    const crisis: Crisis = {
+      id: 1,
+      name: 'Dragon Burning Cities',
+    };
+    await result.navigate(crisisCenterPath);
+
+    fireEvent.click(screen.getByText(crisis.name));
+
+    expect(appLocation.path()).toBe(crisisCenterPath + '/' + crisis.id);
+  });
+});
+```
+
+Alternatively, you can use Spectacular's feature-aware location service,
+[`SpectacularFeatureLocation`](../api/classes/spectacularfeaturelocation):
+
+```ts {3,24-26,29,41}
+import {
+  SpectacularAppComponent,
+  SpectacularFeatureLocation,
+  SpectacularFeatureTestingModule,
+} from '@ngworker/spectacular';
+import {
+  fireEvent,
+  render,
+  RenderResult,
+  screen,
+} from '@testing-library/angular';
+import {
+  Crisis,
+  CrisisCenterModule,
+  crisisCenterPath,
+} from '@tour-of-heroes/crisis-center';
+
+describe('Tour of Heroes: Crisis center integration tests', () => {
+  beforeEach(async () => {
+    result = await render(SpectacularAppComponent, {
+      excludeComponentDeclaration: true,
+      imports: [
+        SpectacularFeatureTestingModule.withFeature({
+          featureModule: CrisisCenterModule,
+          featurePath: crisisCenterPath,
+        }),
+      ],
+    });
+    featureLocation = result.debugElement.injector.get(
+      SpectacularFeatureLocation
+    );
+  });
+
+  let featureLocation: SpectacularFeatureLocation;
+  let result: RenderResult<SpectacularAppComponent, SpectacularAppComponent>;
+
+  it('shows crisis detail when a valid ID is in the URL', async () => {
+    const crisis: Crisis = {
+      id: 1,
+      name: 'Dragon Burning Cities',
+    };
+    await result.navigate(crisisCenterPath);
+
+    fireEvent.click(screen.getByText(crisis.name));
+
+    expect(featureLocation.path()).toBe('~/' + crisis.id);
+  });
+});
+```
+
+## Testing a complete user flow with Spectacular and Angular Testing Library
+
+The following is an example of testing a complete user flow with Spectacular and
+[Angular Testing Library](https://testing-library.com/docs/angular-testing-library/intro/):
+
+```ts {46-64}
+import {
+  SpectacularAppComponent,
+  SpectacularFeatureLocation,
+  SpectacularFeatureRouter,
+  SpectacularFeatureTestingModule,
+} from '@ngworker/spectacular';
+import {
+  fireEvent,
+  render,
+  RenderResult,
+  screen,
+} from '@testing-library/angular';
+import {
+  CrisisCenterModule,
+  crisisCenterPath,
+  CrisisService,
+} from '@tour-of-heroes/crisis-center';
+
+describe('Tour of Heroes: Crisis center integration tests', () => {
+  beforeEach(async () => {
+    result = await render(SpectacularAppComponent, {
+      excludeComponentDeclaration: true,
+      imports: [
+        SpectacularFeatureTestingModule.withFeature({
+          featureModule: CrisisCenterModule,
+          featurePath: crisisCenterPath,
+        }),
+      ],
+    });
+    result.fixture.autoDetectChanges(true);
+    crisisService = result.fixture.debugElement.injector.get(CrisisService);
```

Comment:
Oh this is a nice way of getting the service, I always used the TestBed.
Unrelated, do you think ATL should provide a helper method to do this?

Reply:
I added an inject method to all of Spectacular's test harnesses. So that we mostly don't have to import anything from the Angular testing packages. I expected a helper to be there but this is all I found. There was a reference like result.debugElement but its injector didn't seem to work in the tests I created for this guide.

Reply:
That could be the case, yea.
I haven't tried it before so I'm not 100% sure.
Thanks for the feedback!